### ISSUE TITLE
[redux] API unsetToken Authorization header overwrites default headers

### ISSUE THREAD

**Issue by dennis-tra** [diegohaz/arc#264]
Hi there,

first of all thanks for the great work. 

However I ran into an issue. When my app starts up it checks in the browser storage (window.localStorage) if an access token is available. If it is, it writes it to redux store. If it is not, it runs a "destroySession" function, which does some clean up work (just in case there is some additional data in the local storage). In this function it calls `api.unsetToken()`, which sets the (at this stage empty) header to `{Authorization: undefined}`. Now, when the app sends its first request the `parseSettings` function inside the API service attempts to set the default headers. But since the `unsetToken` function already set a header this has no effect.
 
This is the relevant part of the `parseSettings` function in (src/services/api/index.js):
```javascript
const headers = {
    Accept: "application/json",
    "Content-Type": "application/json",
    "Accept-Language": locale,
  }; 

const settings = {
    body: data ? JSON.stringify(data) : undefined,
    method,
    headers,
    ...otherSettings,
  };
```
`otherSettings` contains `{ headers: { Authorization: undefined } }`, which overwrites the previous headers. My fix at the moment is:
```javascript
const settings = merge({
    body: data ? JSON.stringify(data) : undefined,
    method,
    headers,
  }, otherSettings);
```
(`merge` is already imported and used in the api service)

Will this be a problem in other use cases? I'm happy to file a PR, if you think this is valuable fix.

Thanks for your help

**EDIT 1**: I assume running `setToken` before a first request would result in the same issue?

**Comment by diegohaz** [comment]
You're right. Nice catch, @dennis-tra. 

Would accept a PR. :)

**Comment by dennis-tra** [comment]
cool thanks @diegohaz ! I'll create one :)

**Comment by diegohaz** [comment]
Closed in #265
